pkg/core/hostid/aws.go

```diff
@@ -31,17 +31,11 @@ func AWSUniqueID(cloudMetadataTimeout timeutil.Duration) string {
 
 	client := ec2metadata.New(sess)
 
-	// Checks whether it's an EC2 instance and Metadata service is available.
-	if !client.Available() {
-		log.Debug("No EC2 metadata server detected, assuming not on AWS EC2")
-		return ""
-	}
-
 	doc, err := client.GetInstanceIdentityDocument()
 	if err != nil {
 		log.WithFields(log.Fields{
-			"error": err,
-		}).Error("Failed to get AWS instance-identity")
+			"detail": err,
+		}).Info("No AWS metadata server detected, assuming not on EC2")
```

Comment:
@dloucasfx I maybe missing something but wouldn't this mean even legitimate errors get logged at Info level?

Reply:
correct; the reason we made this change in https://github.com/signalfx/signalfx-agent/pull/1620 because it was a nightmare to debug missing AWSUniqueID unless we set the logs to debug and wait for the issue to be reproduce, on the other hand setting the log to error can raise unwanted alerts, so we settled to info and made the message more descriptive (this applies to all cloud metadata Azure, GCP and AWS)

In addition, in the new SDK code, if I kept `if !client.Available() {` and set that error to info, then we'll be missing the error message

Reply:
Got it, sounds good 👍